packages/cli/testing.ts

```diff
@@ -99,8 +106,29 @@ export async function startPreview({ cwd, command = 'npm run preview' }: Preview
 
 			if (urls && urls.length > 0) {
 				const url = urls[0];
-				resolve({ url, server: process, close });
+				resolve({ url, close });
 			}
 		});
 	});
 }
+
+async function getProcessTree(pid: number) {
+	return new Promise<readonly PS[]>((res, rej) => {
+		pstree(pid, (err, children) => {
+			if (err) rej(err);
+			res(children);
+		});
+	});
+}
+
+async function terminate(pid: number) {
+	const children = await getProcessTree(pid);
+	// the process tree is ordered from parents -> children,
+	// so we'll iterate in the reverse order to terminate the children first
+	for (let i = children.length - 1; i >= 0; i--) {
+		const child = children[i];
+		const pid = Number(child.PID);
+		process.kill(pid);
+	}
+	process.kill(pid);
```

Comment:
Interesting, because this looks exactly like the selling point of [`terminate`](https://www.npmjs.com/package/terminate)

> If you then want to restart your app you naively (we did this!) terminate the "main" process and expect all the "child" processes to be ended too.

> 
> Sadly this results in [Zombie Processes](https://stackoverflow.com/questions/27381163) which you would have to either manually kill or restart the machine/VM to clear.
Instead you need to find all the Process IDs for the child processes that your app created and terminate those before you can re-start your app.

> 
> Using terminate you no longer have this problem.

But whatever works is great!